v1/ast/parser.go

```diff
@@ -26,6 +26,12 @@ import (
 	"github.com/open-policy-agent/opa/v1/ast/location"
 )
 
+// DefaultMaxParsingRecursionDepth is the default maximum recursion depth for the parser
+const DefaultMaxParsingRecursionDepth = 100000
```

Comment:
I'm thinking this could be a variable instead, set by `init()`. We could set it to a much lower value in the tests. It would allow setting the limits much lower for certain architetures, lets say WASM for example.

Reply:
Done!